recipe-templates/corrfunc/common.mk.in

```diff
@@ -338,164 +306,195 @@ ifeq ($(DO_CHECKS), 1)
             endif #Darwin
-          endif #Apple clang. If at some point Apple clang supports OpenMP, then there will need to be an else above this endif. 
+          endif #Apple clang. If at some point Apple clang supports OpenMP, then there will need to be an else above this endif.
         else
           # I dislike being warned multiple times but the compiler warning will not
-          # be visible if the entire codebase is being compiled. 
-          # export WARNING_PRINTED ?= 0
-	        # ifeq ($(WARNING_PRINTED), 0)
-          $(warning $(ccmagenta) $$CC = ${CC} does not support OpenMP - please use gcc/icc for compiling with openmp. Removing $(ccblue)"USE_OMP"$(ccmagenta) from compile options. $(ccreset))
-          infovar := "OPT:=$$(filter-out -DUSE_OMP,$$(OPT))"
-          $(info If you are sure your version of $(ccblue)"clang"$(ccreset) ($(ccblue) must be >= 3.7, NOT Apple clang$(ccreset)) does support OpenMP, then comment out the line $(ccred) $(infovar) $(ccmagenta) in the file $(ccgreen)"common.mk"$(ccreset))
-          $(info You might have to add in the include path (path to $(ccblue)"omp.h"$(ccreset)) to $(ccblue)"CFLAGS"$(ccreset) and the runtime library path to $(ccblue)"CLINK"$(ccreset) at the top of $(ccgreen)"common.mk"$(ccreset))
-
-          # comment out the following line if your version of clang definitely supports OpenMP
-          OPT:=$(filter-out -DUSE_OMP,$(OPT))
-          # export WARNING_PRINTED := 1
-         endif # CLANG_OMP_AVAIL is not 1
+          # be visible if the entire codebase is being compiled.
+          export WARNING_PRINTED ?= 0
+          ifeq ($(WARNING_PRINTED), 0)
+            $(warning $(ccmagenta) $$CC = ${CC} does not support OpenMP - please use gcc/icc for compiling with openmp. Removing $(ccblue)"USE_OMP"$(ccmagenta) from compile options. $(ccreset))
+            infovar := "OPT:=$$(filter-out -DUSE_OMP,$$(OPT))"
+            $(info If you are sure your version of $(ccblue)"clang"$(ccreset) ($(ccblue) must be >= 3.7, NOT Apple clang$(ccreset)) does support OpenMP, then comment out the line $(ccred) $(infovar) $(ccmagenta) in the file $(ccgreen)"common.mk"$(ccreset))
+            $(info You might have to add in the include path (path to $(ccblue)"omp.h"$(ccreset)) to $(ccblue)"CFLAGS"$(ccreset) and the runtime library path to $(ccblue)"CLINK"$(ccreset) at the top of $(ccgreen)"common.mk"$(ccreset))
+
+            # comment out the following line if your version of clang definitely supports OpenMP
+            OPT:=$(filter-out -DUSE_OMP,$(OPT))
+            export WARNING_PRINTED := 1
+          endif
+        endif # CLANG_OMP_AVAIL is not 1
       endif # USE_OMP
     endif # CC is clang
 
-    # #### common options for gcc and clang
-    # ifeq (USE_AVX,$(findstring USE_AVX,$(OPT)))
-    #   CFLAGS  +=  -mavx
-    # endif
-
     CFLAGS += -funroll-loops
-    # CFLAGS += -march=native -fno-strict-aliasing
+    CFLAGS += -march=native -fno-strict-aliasing
```

Comment:
@lgarrison @beckermr @manodeep

This (-march=native) may be the reason why we are producing less compatible builds. If the machine supports avx2, would -march-native enable avx2 automatically?

Reply:
Yup - `-march=native` will not produce a portable build. This will need a bit of a think to produce optimised *and* portable builds

Reply:
Yeah, I think the immediate solution is to remove `-march=native`, and possibly replace it with `-march=ivybridge`, which is the last architecture before AVX2 was introduced.

Unfortunately, this will lose the benefits of the AVX2 and AVX-512 kernels on platforms that support them.  I think a general solution to this problem is difficult, and might require changes to the Corrfunc build structure.  One approach is to build each version of the kernel in a separate compilation unit and link them all together at the end; another would be to build multiple versions of the whole library and let the calling code choose the right version.

Or if conda has some mechanism to build and distribute different binaries for platforms with different instruction sets, that might be easier.

Reply:
So conda-forge limits its SIMD optimizations to nocona. I had thought I had put in a patch in the feedstock to turn off native, but I guess I did not. 

https://github.com/conda-forge/corrfunc-feedstock

Reply:
My advice would be to leave corrfunc as is and patch the feedstock to turn off native complilation

Reply:
Yes unfortunately conda doesn't have such refined per arch binary targets -- yet for data science FLOPS is getting more important.
conda more or less has that ability (the linux-64 tag can be refined to track different cpu archs). Would be nice if somebody(e.g. us?) actually starts making use of that.